pkgs/development/libraries/glibc/locales.nix

```diff
@@ -19,19 +19,26 @@ callPackage ./common.nix { inherit stdenv; } {
 
   extraNativeBuildInputs = [ glibc ];
 
-  # Awful hack: `localedef' doesn't allow the path to `locale-archive'
-  # to be overriden, but you *can* specify a prefix, i.e. it will use
-  # <prefix>/<path-to-glibc>/lib/locale/locale-archive.  So we use
-  # $TMPDIR as a prefix, meaning that the locale-archive is placed in
-  # $TMPDIR/nix/store/...-glibc-.../lib/locale/locale-archive.
-  buildPhase =
-    ''
-      mkdir -p $TMPDIR/"${buildPackages.glibc.out}/lib/locale"
+  LOCALEDEF_FLAGS = [
+    # Awful hack: `localedef' doesn't allow the path to `locale-archive'
+    # to be overriden, but you *can* specify a prefix, i.e. it will use
+    # <prefix>/<path-to-glibc>/lib/locale/locale-archive.  So we use
+    # /tmp as a prefix, meaning that the locale-archive is placed in
+    # /tmp/nix/store/...-glibc-.../lib/locale/locale-archive.
+    "--prefix=/tmp"
```

Comment:
Alternatively you could keep `--prefix='$TMPDIR'` inlined below and have here only static `--{little,big}-endian` flags appended via an extra variable. That way you could shrink the diff down to 2 lines and make it less scary :)

On `locale-archive` portability. My understanding is that it's format is optimised for a direct `mmap()` from disk to be readily usable. It's `struct __locale_data` is the following: https://sourceware.org/git/?p=glibc.git;a=blob;f=locale/localeinfo.h;h=fd43033a1915a0356ce6b776a4c8fead4866086c;hb=HEAD#l49

Do you know if is careful enough to generate valid locale archive for 32-bit system from 64-bit system? Probably the best test would be from `x86_64` (64-bit LE, 4-byte `int` natural alignment) to `m68k` (32-bit BE, 2-byte unusual `int` alignment) and run cross-compiled `hello` under `qemu-m68k`.

Reply:
So, from my debugging, I saw that the failed load check came from mmaping the locale-archive header, whose struct is defined here: https://sourceware.org/git/?p=glibc.git;a=blob;f=locale/locarchive.h;h=e76e7ce185badcb2b3def1a5b5179c8a283a93d0;hb=HEAD#l27

I'm not completely sure whether the locale generation is 32/64 bit agnostic, I'll do some testing.

Reply:
Oh, your link looks better! Let's declare it done.